### ISSUE TITLE
[BUG] Can't exec `bin` that start with `.`

### ISSUE THREAD

**Issue by remiposo** [npm/package-json#116]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Current Behavior

After #105, `bin` that start with `.` are now considered invalid to prevent out of scope path traversals.　As a result, `bin` like `"bin": {"hoge": ".bin/hoge.js"}` are now also considered invalid and removed. Was this an intentional change?

It seems that scripts like `create-rust-webpack` are no longer executable.
https://github.com/rustwasm/rust-webpack-template/issues/190

```sh
❯ npm init rust-webpack
npm error could not determine executable to run
```

### Expected Behavior

If this change was not intentional, I would like to continue executing scripts as before.

### Steps To Reproduce

1. Run `npm init rust-webpack`

### Environment

- npm: 10.8.2
- Node: 22.6.0
- OS: Sonoma 14.5
- platform: M2 Mac mini
